src/main/java/org/jnode/fs/ntfs/FileRecord.java

```diff
@@ -132,7 +134,7 @@ public void checkIfValid() throws IOException {
         // check for the magic number to see if we have a file record
         if (getMagic() != Magic.FILE) {
             if (log.isDebugEnabled()) {
-                log.debug("Invalid magic number found for FILE record: {} -- dumping buffer", getMagic());
+                log.debug("Invalid magic number found for FILE record({}): {} -- dumping buffer", referenceNumber, getMagic());
```

Comment:
I tried to improve the logging here by adding in the MFT index so that you know which record is triggering read operations, etc